### ISSUE TITLE
Trying to keep multiple resolutions of a movie

### ISSUE THREAD

**Issue by Peiloy** [Radarr/Radarr#1910]
For a few movies I would like to keep a 1080P version and a 4K Version. When I download the 4K version the 1080 deletes and vice versa. Is there any way to stop Radarr from deleting a resolution I want to keep?

**Comment by hotio** [comment]
No, has been asked multiple times already

**Comment by drashna** [comment]
All due respect, not including this ability in Radarr is stupid, and a huge oversight.  

Plex and Emby support multiple resolutions/formats in the same folder.  And will automatically grab the most appropriate file, and reduce the likelihood that files need to be transcoded. 

While I understand that most won't need this, for those of us that would like it, maybe keep it as an advanced setting that is off by default.

**Comment by hotio** [comment]
Create a pull request for this, you can even collaborate with the few people that want this. Radarr is opensource and the devs would be happy to merge this PR I think.

**Comment by galli-leo** [comment]
@drashna If it would be as easy as just adding a toggle, we would have done it a long time ago. But because this is built on Sonarr (and Sonarr was made with one file per episode in mind) it's not that easy.

**Comment by woble** [comment]
is there an option to not remove previously downloaded file? Perhaps that can be poor man's solution to download and keep multiple versions.

**Comment by micdah** [comment]
Yeah having the option to not delete lower/existing quality files, would be a good addition. 

Couchpotato has this feature already, but it is cumbersome to manual download multiple qualities using Couchpotato - so a switch to Radarr would be ideal with the addition of this feature.

A way to keep around 2160p and 1080p versions, even though one would have to manually search for the alternate quality, but at least it would be possible to do.

**Comment by woble** [comment]
It could be per profile where each selected quality can be set to be removed or kept when a new quality is downloaded. Cutoff quality would naturally ignore that setting, or perhaps the checkbox for the cutoff quality shouldn't even be shown.

![image](https://user-images.githubusercontent.com/494873/34227855-61ca68a4-e61a-11e7-9df8-c39dbc0e0bd0.png)

**Comment by AnnieTheEagle** [comment]
This would be a very nice feature. Being able to keep a very high quality version for when you're on your fiber optic at home, and then a lower bit rate release for when you're on dubious hotel Internet. 

I think there would need to be three changes:
1. A setting on whether or not to keep multiple versions
2. On import checking this setting and deciding whether or not to delete existing version because of this setting
3. Disk Scanner returns multiple files if present (I believe right now it returns the first, in filename order)

Personally I think having this would be great. We already have the UI since the tab itself says "Files" not "File" and has more than enough space to display multiple versions.

This quick fix won't allow for multiple Bluray-1080p since they'd have the same file name. But rather allow for someone to keep a Remux-2160p with a Bluray-1080p

**Comment by intelligence** [comment]
This sure sound interesting. I'm having issues with 4K content as my server can't transcode it on the fly. And having Plex converting it is less than ideal since you can't really tweak the conversion settings that well.

**Comment by jalmansor** [comment]
This would even be useful in Sonarr, for the same reasons. Wonder if it's something that might be in their v3 work...

**Comment by jalmansor** [comment]
Well, there's at least an open issue, though it seems to be of equally low priority https://github.com/Sonarr/Sonarr/issues/312

**Comment by donkeyjuks** [comment]
any update to this?... I'd also like to keep 4K version and 1080p version (different folders).

**Comment by galli-leo** [comment]
@donkeyjuks https://blog.radarr.video/development/update/2018/11/11/roadmap-update.html

**Comment by mattbailey** [comment]
My super lazy hack: https://github.com/mattbailey/Radarr/commit/f9300900c2ca2410b8f7d5875a0727ccc4ba2b5f

**Comment by ravensorb** [comment]
Maybe we at least have a setting to enable/disable deleting existing videos?

**Comment by epheterson** [comment]
Could we re-visit this? I figured I was just missing a setting but this would be a great feature to add.

**Comment by Qstick** [comment]
It’s a tad more complicated than that if we want to do it right.

**Comment by clem16** [comment]
I came looking to see if this was a thing yet. I realize it's not super easy to implement, yet I'm looking forwards to it being available.
I usually like to keep multiple encodes around for local streaming - laptop etc and then higher quality encodes or REMUX's etc for streaming directly to larger screen tv's.

The quick'n dirty hack that I've been considering is simply running multiple versions of radarr in separate jails, each managing their own quality - tho in my opinion this is like taking a buzz saw to a loaf of bread. It works, but it's messy and you end up with bit's n pieces everywhere.

**Comment by Puntoboy** [comment]
When running two copies of Radarr, do you need two separate media folders where the movies are stored or can you have one location just with two versions of Radarr writing to it?

**Comment by austinwbest** [comment]
> When running two copies of Radarr, do you need two separate media folders where the movies are stored or can you have one location just with two versions of Radarr writing to it?

Each instance has its own root folders & you can just join them in plex for example

**Comment by azumukupoe** [comment]
Could use this for movies that are split into two parts

**Comment by bakerboy448** [comment]
> Could use this for movies that are split into two parts

not really, it’s a different beast with its own hurdlers and logic.

this request is two discrete versions in each a single file.  1 movie -> many versions; One version -> One File

what you’re saying would be multiple versions and multiple files

1 movie -> many versions; many versions -> many files

significantly more complex

**Comment by azumukupoe** [comment]
> significantly more complex

should I open a separate issue or is there any i missed?

**Comment by ryanfb** [comment]
@azumukupoe you're looking for #145

**Comment by timmehtimtims** [comment]
> > When running two copies of Radarr, do you need two separate media folders where the movies are stored or can you have one location just with two versions of Radarr writing to it?
> 
> Each instance has its own root folders & you can just join them in plex for example

I guess what I'm finding is that having multiple instances takes up a huge amount of HDD space. I have a 500Gb nvme and each instance is about 25Gb, using up valuable space. I have 3 instances: 4K, 1080p REMUX and low quality Web-DL 1080p/720p = 75Gb (nearly 1/5th) of my HDD, just to run radarr. If there was a way to do this in once instance then that would be amazing. 👍

**Comment by Puntoboy** [comment]
Having one instance wouldn't make a difference in your case, the video files will still take up that space. 

In that situation the best way is to have a single best quality version (4K) and then transcode it on the fly for the devices that need it. That just takes a lot of CPU/GPU to do.

**Comment by timmehtimtims** [comment]
> Having one instance wouldn't make a difference in your case, the video files will still take up that space.
> 
> In that situation the best way is to have a single best quality version (4K) and then transcode it on the fly for the devices that need it. That just takes a lot of CPU/GPU to do.

Sorry, I should have been more clear - the metadata in /opt/appdata/radarr is 25Gb. Same again for /opt/appdata/radarr4k and /opt/appdata/radarrlowquality
One instance would mean I'd reduce the amount of space used by metadata by 1/3.

**Comment by Puntoboy** [comment]
How big if your library? I have over 500 movies and the metadata on mine is only 700mb each.

**Comment by timmehtimtims** [comment]
Haha, good point. Maybe this isn't a common issue...

**Comment by epheterson** [comment]
This is still desirable, please consider!

**Comment by ravensorb** [comment]
Agreed!

**Comment by bouncy99** [comment]
Agreed, this is still desirable, and yet lacking for the last 4 years :(

**Comment by timmehtimtims** [comment]
Again, agreed desirable :)

**Comment by ta264** [comment]
Please, comments like this are not helpful. We know it's wanted, but it is *a lot* of work.

**Comment by ravensorb** [comment]
Question - is there a discussion around ideas on how to implement?  Its possible others might be able to help if there is kind of an agreed approach.

**Comment by bakerboy448** [comment]
> Question - is there a discussion around ideas on how to implement? Its possible others might be able to help if there is kind of an agreed approach.

Some rough initial discussion is taking place over on Sonarr
https://github.com/Sonarr/Sonarr/issues/4551

**Comment by ta264** [comment]
And just to be clear, constructive suggestions/ discussion is very welcome. It's only "+1" and "I can't believe this hasn't been done in 4 years" that are to be avoided.

**Comment by MacGyver27** [comment]
@ta264 Can you please explain why that would be lot of work? On the development level please. Thanks

**Comment by austinwbest** [comment]
> @ta264 Can you please explain why that would be lot of work? On the development level please. Thanks

Because the entire app is setup for 1:1

The UI, the api, the database, the code base. Everything has to be switched from that to a 1:many and a lot of the database has to be rearranged to accomadate that as well. The grab logic, upgrade, rss, cutoff, folder structures, etc is all 1:1

Everything is going from a single string checks to list loops for example.

**Comment by kaysond** [comment]
Does radarr distinguish between regular files and symlinks? Meaning could I have separate radarr instances for 1080p/4k and symlink video files from one into the folders of another? It would be a decent workaround that supports multiple versions in emby/jellyfin

**Comment by lunks** [comment]
My 2c on two ways to implement this in a way that could help some scenarios: Allow us to lock or ignore the file for Radarr i.e. for all intents and purposes the file would be either locked or ignored. If file is locked, it would never try to upgrade or remove the locked file. If ignoring, Radarr could keep doing its thing for the movie but would solely ignore the file that has been marked as ignored. Some examples:

## Considering multiple resolutions:


### If lock is the way to go:
I have one 1080p file and a 4k file.

I decide that the 1080p is great and I prefer it since it's more compatible so I lock it. Radarr can eventually get rid of the 4k given it's not locked and a locked file exists, so it is free to do whatever with other files that are not locked.

I want to keep both, I lock both which would behave similar as if I'm not monitoring given movie. It can't rename, can't upgrade, can't remove any of them.

### If ignore is the way to go:
I have one 1080p file and a 4k file.

I decide that 1080p is great and I would like to keep it because the 4k version needs transcoding. I set radarr to ignore the 1080p file and so Radarr is free to upgrade the 4k file, replace it, etc.

If I decide that I want to keep both versions forever, ignoring both would make Radarr consider those files are not there, downloading a third version.

## Considering multiple versions i.e. theatrical and extended:

### If lock is the way to go:

I love this particular movie theatrical version and it's a rare find, I manually downloaded it outside radarr, added the file and locked it. I'll only have this theatrical version because the file is locked. If I want to add an extended version, I'd have to manually add that file and mark it as locked. Radarr would never upgrade these files.

### If ignore is the way to go:

Same theatrical version that is hard as hell to find. I set the file as ignored, then Radarr won't ever mess with it. It could download an extended version in better quality, a theatrical version in better quality even, but the one I ripped and that gave me a lot of work would be kept there.

**Comment by cjemorton** [comment]
I seriously like the approach of locking a file.

Say, for example.  There's encodes I have personally done for a show but
there's also 4k releases that are out. I don't want to nix the DVD encode,
for playing on smaller devices. But having the ability to grab a 4k release
for streaming to larger TV that support it.

Being able to lock all files for a release group. (I tag all my personal
encodes with my own tag), then if a higher quality comes out, or is
obtained,  it gets imported alongside the locked encode.

**Comment by ta264** [comment]
I don't think anything that requires a manual intervention for every case is a viable solution.

If and when we do something about this, it will be a something that allows upgrades via profiles I think. We don't have much interest in a half-baked stop-gap solution - we should do it right or not do it at all in my view.

**Comment by ptyork** [comment]
Is there not a _relatively_ easy, non-breaking option that doesn't involve breaking the 1:1 ImdbId:MovieId relationship? In looking (admittedly very briefly) at the model files, it seems maybe you could add a MovieVersion class:
```
        public bool Monitored { get; set; }    // MAYBE...if need to monitor separately
        public string Path { get; set; }   // MAYBE...if multiple versions in 1 folder is a major issue...could also just append profile name to the parent movie folder
        public Profile Profile { get; set; }
        public int ProfileId { get; set; }
        public HashSet<int> Tags { get; set; }
        public MovieFile MovieFile { get; set; }
        public int MovieFileId { get; set; }
```
Then modify Movie to include:
```
        public bool AdditionalVersions { get; set; }
        public List<MovieVersion> AdditionlVersions { get; set; }
```
Then basically in the UI you'd check a box and be able to add additional profiles. So you'd have a primary profile (non-breaking) and alternate profiles. Since profiles include both quality and custom formats, you'd be able to handle both the multiple quality and multiple languages, editions, cuts, etc. requests. Other metadata like Runtime, Year, a VersionTitle may also be needed/wanted. But you get the idea.

Obviously oversimplified and there's a decent amount of work to actually implement this since it touches so many bits of system---movie lists, indexers, importers, parsers, etc. But it would seem to at least avoid breaking changes to the core Movie model (really adding only a bool to the table) and relying on separate tables for these edge cases). And would not over complicate the UI as the changes would be hidden in a check-to-show container. Feels like a major effort but not a ground-up rewrite.

Just a thought form a naive observer.

**Comment by Nakda** [comment]
I simply want to contribute with my requirements related to this feature:
- I have got several users connected to my Plex server, and they add movies through a request management system (Overseerr, Petio, Ombi, etc.) connected to Radarr
- Some users prefer movies in their original audio language, others prefer our local (french) language
- In addition to user preferences, it also depends on the movie genre (Animation movies usually preferred in french in all cases for example)

At the moment, the only solution is to run two Radarr instances, with two folder structures, and probably two instances of a request management system.

What I think would best answer our need would be the ability to request multiple Quality Profiles for a movie monitored by Radarr:
- When adding a new movie, I can select multiple profiles (French 1080p, Original 4K, Original 1080p, etc.)
- When requesting movies from a request management system, the requester will be able to select the profiles they want to see added because they can be fetched from Radarr
- If a movie is already available as the "French 1080p" version, I can request the "French 4K" or any other

The "Keep" toggle [suggested above](https://github.com/Radarr/Radarr/issues/1910#issuecomment-353178603) probably won't suit my needs since we might not always want a 1080p and a 4K version. I think it would be best for this to stay flexible.

Does that make sense?

**Comment by bakerboy448** [comment]
@Nakda hoe would you handle if two different quality profiles overlap and want to download the same movie?

As far as the keep toggle comments, sounds like that's a use case for the existing process.

**Comment by Nakda** [comment]
> @Nakda hoe would you handle if two different quality profiles overlap and want to download the same movie?
> 
> As far as the keep toggle comments, sounds like that's a use case for the existing process.

I can't think of a scenario like this, but maybe it's because I'm using Quality Profiles in a different way. Do you have an example in mind?

**Comment by ClemTheStudent** [comment]
> It could be per profile where each selected quality can be set to be removed or kept when a new quality is downloaded. Cutoff quality would naturally ignore that setting, or perhaps the checkbox for the cutoff quality shouldn't even be shown.
> 
> ![image](https://user-images.githubusercontent.com/494873/34241291-3787bc22-e660-11e7-883a-0ddef7fad3e4.png)
Hi,
Has anything been done since?
